--- gnash-6675d736bf4d78e7ece0879b98d0b19accb9611d/libmedia/ffmpeg/VideoDecoderFfmpegVaapi.cpp	2015-03-17 00:10:12.000000000 +0900
+++ gnash/libmedia/ffmpeg/VideoDecoderFfmpegVaapi.cpp	2015-05-12 12:44:09.591057502 +0900
@@ -46,20 +46,20 @@
 
     const int *profiles;
     switch (codec_id) {
-    case CODEC_ID_MPEG2VIDEO:
+    case AV_CODEC_ID_MPEG2VIDEO:
         profiles = mpeg2_profiles;
         break;
-    case CODEC_ID_MPEG4:
-    case CODEC_ID_H263:
+    case AV_CODEC_ID_MPEG4:
+    case AV_CODEC_ID_H263:
         profiles = mpeg4_profiles;
         break;
-    case CODEC_ID_H264:
+    case AV_CODEC_ID_H264:
         profiles = h264_profiles;
         break;
-    case CODEC_ID_WMV3:
+    case AV_CODEC_ID_WMV3:
         profiles = wmv3_profiles;
         break;
-    case CODEC_ID_VC1:
+    case AV_CODEC_ID_VC1:
         profiles = vc1_profiles;
         break;
     default:
