diff -ur --unidirectional-new-file -u -r macuim.orig/unix/uim/mozc.cc macuim/unix/uim/mozc.cc
--- macuim.orig/unix/uim/mozc.cc	2015-12-07 05:18:52.193220649 +0000
+++ macuim/unix/uim/mozc.cc	2015-12-07 05:20:18.367229217 +0000
@@ -48,6 +48,8 @@
 # include "plugin.h"
 #endif
 
+#include "base/init_mozc.h"
+
 #include "base/port.h"
 #include "base/util.h"
 #include "protocol/config.pb.h"
@@ -1196,7 +1198,7 @@
   argv[0] = (char *)name;
   argv[1] =  NULL;
 
-  InitGoogle((const char *)argv[0], &argc, (char ***)&argv, true);
+  mozc::InitMozc((const char *)argv[0], &argc, (char ***)&argv, true);
   mozc::uim::install_keymap();
 }
 
