--- meld/meldapp.py.orig	2015-11-08 17:43:48.067051476 +0000
+++ meld/meldapp.py	2015-11-08 17:52:31.354990868 +0000
@@ -98,6 +98,8 @@
     def do_window_removed(self, widget):
         widget.meldwindow = None
         Gtk.Application.do_window_removed(self, widget)
+        if not len(self.get_windows()):
+            self.quit()
 
     # We can't override do_local_command_line because it has no introspection
     # annotations: https://bugzilla.gnome.org/show_bug.cgi?id=687912
