diff -ur --unidirectional-new-file -ur uim-9999.orig/qt4/immodule/plugin.cpp uim-9999/qt4/immodule/plugin.cpp
--- uim-9999.orig/qt4/immodule/plugin.cpp	2016-01-17 23:41:46.618090279 +0000
+++ uim-9999/qt4/immodule/plugin.cpp	2016-01-17 23:43:09.636088356 +0000
@@ -43,6 +43,7 @@
 # endif
 # include <QtGui/QInputContext>
 #else
+# include <QtCore/qdatastream.h>
 # if defined(Q_OS_UNIX) && !defined(Q_OS_MAC)
 #  include <QtX11Extras/QX11Info>
 # endif
