diff -ur --unidirectional-new-file -urN a/examples/Makefile.am b/examples/Makefile.am
--- a/examples/Makefile.am	2019-03-24 11:18:36.248953107 +0000
+++ b/examples/Makefile.am	2019-03-24 11:19:58.515137413 +0000
@@ -1,7 +1,7 @@
-docsdir = $(datadir)/doc/$(PACKAGE)
+docsdir = $(datadir)/doc/$(PACKAGE)-9999
 docs_DATA = README
 
-examplesdir = $(datadir)/doc/$(PACKAGE)/examples
+examplesdir = $(datadir)/doc/$(PACKAGE)-9999/examples
 examples_DATA =					\
 	background-image.py			\
 	block-size-column.py		\
