diff -ur --unidirectional-new-file -urN webkitgtk-2.10.7.orig/Source/JavaScriptCore/runtime/Options.cpp webkitgtk-2.10.7/Source/JavaScriptCore/runtime/Options.cpp
--- webkitgtk-2.10.7.orig/Source/JavaScriptCore/runtime/Options.cpp	2016-03-08 21:08:57.992140240 +0000
+++ webkitgtk-2.10.7/Source/JavaScriptCore/runtime/Options.cpp	2016-03-08 21:10:38.310142170 +0000
@@ -610,7 +610,7 @@
     case Options::Type::unsignedType:
         return m_entry.unsignedVal == other.m_entry.unsignedVal;
     case Options::Type::doubleType:
-        return (m_entry.doubleVal == other.m_entry.doubleVal) || (isnan(m_entry.doubleVal) && isnan(other.m_entry.doubleVal));
+        return (m_entry.doubleVal == other.m_entry.doubleVal) || (std::isnan(m_entry.doubleVal) && std::isnan(other.m_entry.doubleVal));
     case Options::Type::int32Type:
         return m_entry.int32Val == other.m_entry.int32Val;
     case Options::Type::optionRangeType:
