diff --git a/launcher/game/project.rpy b/launcher/game/project.rpy
index aa9fa9f29..13aa5ec2a 100644
--- a/launcher/game/project.rpy
+++ b/launcher/game/project.rpy
@@ -548,6 +548,9 @@ init python in project:
                 elif os.path.exists(os.path.join(p.path, "game", "tl", _preferences.language)):
                     rv = p
 
+                elif not os.path.exists(os.path.join(rv.path, "game", "tl", _preferences.language)):
+                    rv = p
+
             self.tutorial_language = language
             self.tutorial = rv
