diff -ur --unidirectional-new-file -urN renpy-6.99.14.3-source/renpy/common/_compat/styles.rpym renpy.new/renpy/common/_compat/styles.rpym
--- renpy-6.99.14.3-source/renpy/common/_compat/styles.rpym	2018-01-07 23:47:18.000000000 +0000
+++ renpy.new/renpy/common/_compat/styles.rpym	2018-04-19 22:00:21.657255517 +0000
@@ -352,13 +352,6 @@
             self.__dict__ = { "target" : self.target }
 
 
-    style.selected_button = _SelectedCompat('button')
-    style.selected_button_text = _SelectedCompat('button_text')
-    style.gm_nav_selected_button = _SelectedCompat('gm_nav_button')
-    style.gm_nav_selected_button_text = _SelectedCompat('gm_nav_button_text')
-    style.prefs_selected_button = _SelectedCompat('prefs_button')
-    style.prefs_selected_button_text = _SelectedCompat('prefs_button_text')
-
     def _apply_selected_compat():
         for scs in _selected_compat:
             target = getattr(style, scs.target)
