diff -ur --unidirectional-new-file -u -rN xorg-server-1.18.2.orig/Xext/xf86bigfont.c xorg-server-1.18.2/Xext/xf86bigfont.c
--- xorg-server-1.18.2.orig/Xext/xf86bigfont.c	2016-03-15 06:01:31.768050070 +0000
+++ xorg-server-1.18.2/Xext/xf86bigfont.c	2016-03-15 06:15:53.547066648 +0000
@@ -47,9 +47,7 @@
    PAGE_SIZE. It is defined in <asm/page.h>. */
 #include <asm/page.h>
 #endif
-#ifdef SVR4
 #include <sys/sysmacros.h>
-#endif
 #if defined(__CYGWIN__)
 #include <sys/param.h>
 #include <sys/sysmacros.h>
diff -ur --unidirectional-new-file -u -rN xorg-server-1.18.2.orig/config/udev.c xorg-server-1.18.2/config/udev.c
--- xorg-server-1.18.2.orig/config/udev.c	2016-03-15 06:01:31.787050070 +0000
+++ xorg-server-1.18.2/config/udev.c	2016-03-15 06:14:49.588065418 +0000
@@ -31,6 +31,8 @@
 #include <ctype.h>
 #include <unistd.h>
 
+#include <sys/sysmacros.h>
+
 #include "input.h"
 #include "inputstr.h"
 #include "hotplug.h"
diff -ur --unidirectional-new-file -u -rN xorg-server-1.18.2.orig/hw/xfree86/common/xf86Xinput.c xorg-server-1.18.2/hw/xfree86/common/xf86Xinput.c
--- xorg-server-1.18.2.orig/hw/xfree86/common/xf86Xinput.c	2016-03-15 06:01:31.801050070 +0000
+++ xorg-server-1.18.2/hw/xfree86/common/xf86Xinput.c	2016-03-15 06:13:42.011064118 +0000
@@ -84,6 +84,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <unistd.h>
+#include <sys/sysmacros.h>
 #ifdef HAVE_SYS_MKDEV_H
 #include <sys/mkdev.h>          /* for major() & minor() on Solaris */
 #endif
diff -ur --unidirectional-new-file -u -rN xorg-server-1.18.2.orig/hw/xfree86/os-support/linux/lnx_init.c xorg-server-1.18.2/hw/xfree86/os-support/linux/lnx_init.c
--- xorg-server-1.18.2.orig/hw/xfree86/os-support/linux/lnx_init.c	2016-03-15 06:01:31.805050070 +0000
+++ xorg-server-1.18.2/hw/xfree86/os-support/linux/lnx_init.c	2016-03-15 06:13:02.925063366 +0000
@@ -38,6 +38,7 @@
 #include "xf86_OSlib.h"
 
 #include <sys/stat.h>
+#include <sys/sysmacros.h>
 
 #ifndef K_OFF
 #define K_OFF 0x4
diff -ur --unidirectional-new-file -u -rN xorg-server-1.18.2.orig/hw/xfree86/os-support/xf86_OSlib.h xorg-server-1.18.2/hw/xfree86/os-support/xf86_OSlib.h
--- xorg-server-1.18.2.orig/hw/xfree86/os-support/xf86_OSlib.h	2016-03-15 06:01:31.806050070 +0000
+++ xorg-server-1.18.2/hw/xfree86/os-support/xf86_OSlib.h	2016-03-15 06:15:58.823066749 +0000
@@ -113,14 +113,12 @@
 #endif
 #endif                          /* _NEED_SYSI86 */
 
-#if defined(HAS_SVR3_MMAPDRV)
 #include <sys/sysmacros.h>
 #if !defined(_NEED_SYSI86)
 #include <sys/immu.h>
 #include <sys/region.h>
 #endif
 #include <sys/mmap.h>           /* MMAP driver header */
-#endif
 
 #if !defined(sun) || defined(HAVE_SYS_VT_H)
 #define HAS_USL_VTS
