diff -u -r pinentry-0.9.6.orig/m4/qt.m4 pinentry-0.9.6/m4/qt.m4
--- pinentry-0.9.6.orig/m4/qt.m4	2015-09-11 03:21:54.000000000 +0900
+++ pinentry-0.9.6/m4/qt.m4	2015-09-15 22:31:58.325000000 +0900
@@ -28,43 +28,6 @@
 
 AC_DEFUN([FIND_QT],
 [
-  PKG_CHECK_MODULES(PINENTRY_QT,
-                    Qt5Core >= 5.0.0 Qt5Gui >= 5.0.0 Qt5Widgets >= 5.0.0,
-                    [have_qt5_libs="yes"],
-                    [have_qt5_libs="no"])
-
-  if "$PKG_CONFIG" --variable qt_config Qt5Core | grep -q "reduce_relocations"; then
-    PINENTRY_QT_CFLAGS="$PINENTRY_QT_CFLAGS -fpic"
-  fi
-
-  if test "$have_qt5_libs" = "yes"; then
-    AC_CHECK_TOOL(MOC, moc)
-    AC_MSG_CHECKING([moc version])
-    mocversion=`$MOC -v 2>&1`
-    mocversiongrep=`echo $mocversion | grep "Qt 5\|moc 5"`
-    if test x"$mocversiongrep" != x"$mocversion"; then
-      AC_MSG_RESULT([no])
-      # moc was not the qt5 one, try with moc-qt5
-      AC_CHECK_TOOL(MOC2, moc-qt5)
-      mocversion=`$MOC2 -v 2>&1`
-      mocversiongrep=`echo $mocversion | grep "Qt 5\|moc-qt5 5\|moc 5"`
-      if test x"$mocversiongrep" != x"$mocversion"; then
-        AC_CHECK_TOOL(QTCHOOSER, qtchooser)
-        qt5tooldir=`QT_SELECT=qt5 qtchooser -print-env | grep QTTOOLDIR | cut -d '=' -f 2 | cut -d \" -f 2`
-        mocversion=`$qt5tooldir/moc -v 2>&1`
-        mocversiongrep=`echo $mocversion | grep "Qt 5\|moc 5"`
-        if test x"$mocversiongrep" != x"$mocversion"; then
-          # no valid moc found
-          have_qt5_libs="no";
-        else
-          MOC=$qt5tooldir/moc
-        fi
-      else
-        MOC=$MOC2
-      fi
-    fi
-  fi
-  if test "$have_qt5_libs" != "yes"; then
     PKG_CHECK_MODULES(PINENTRY_QT,
                       QtCore >= 4.4.0 QtGui >= 4.4.0,
                       [have_qt4_libs="yes"],
@@ -89,5 +52,4 @@
         fi
       fi
     fi
-  fi
 ])
