diff -ur --unidirectional-new-file -urN 5489402ca1fe17d5b4d20e4dcdc6ac8e6e415b0b.orig/src/backlight.c 5489402ca1fe17d5b4d20e4dcdc6ac8e6e415b0b/src/backlight.c
--- 5489402ca1fe17d5b4d20e4dcdc6ac8e6e415b0b.orig/src/backlight.c	2016-03-15 03:44:51.524022267 +0000
+++ 5489402ca1fe17d5b4d20e4dcdc6ac8e6e415b0b/src/backlight.c	2016-03-15 03:45:55.847023505 +0000
@@ -34,11 +34,7 @@
 #include <sys/stat.h>
 #include <sys/ioctl.h>
 
-#if MAJOR_IN_MKDEV
-#include <sys/mkdev.h>
-#elif MAJOR_IN_SYSMACROS
 #include <sys/sysmacros.h>
-#endif
 
 #include <stdio.h>
 #include <stdlib.h>
diff -ur --unidirectional-new-file -urN 5489402ca1fe17d5b4d20e4dcdc6ac8e6e415b0b.orig/src/intel_device.c 5489402ca1fe17d5b4d20e4dcdc6ac8e6e415b0b/src/intel_device.c
--- 5489402ca1fe17d5b4d20e4dcdc6ac8e6e415b0b.orig/src/intel_device.c	2016-03-15 03:44:51.524022267 +0000
+++ 5489402ca1fe17d5b4d20e4dcdc6ac8e6e415b0b/src/intel_device.c	2016-03-15 03:45:48.765023368 +0000
@@ -38,11 +38,7 @@
 #include <dirent.h>
 #include <errno.h>
 
-#if MAJOR_IN_MKDEV
-#include <sys/mkdev.h>
-#elif MAJOR_IN_SYSMACROS
 #include <sys/sysmacros.h>
-#endif
 
 #include <pciaccess.h>
 
diff -ur --unidirectional-new-file -urN 5489402ca1fe17d5b4d20e4dcdc6ac8e6e415b0b.orig/tools/backlight_helper.c 5489402ca1fe17d5b4d20e4dcdc6ac8e6e415b0b/tools/backlight_helper.c
--- 5489402ca1fe17d5b4d20e4dcdc6ac8e6e415b0b.orig/tools/backlight_helper.c	2016-03-15 03:44:51.541022267 +0000
+++ 5489402ca1fe17d5b4d20e4dcdc6ac8e6e415b0b/tools/backlight_helper.c	2016-03-15 03:45:42.695023252 +0000
@@ -9,11 +9,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 
-#if MAJOR_IN_MKDEV
-#include <sys/mkdev.h>
-#elif MAJOR_IN_SYSMACROS
 #include <sys/sysmacros.h>
-#endif
 
 #define DBG 0
 
