8da42fb5ffff3ae1b152eaa7bc4b98f283d4dc03
diff --git a/CMakeLists.txt b/CMakeLists.txt
index e2b4f23..e9aa2f8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,7 +15,7 @@ pkg_check_modules(FCITX_QT "fcitx-qt>=4.2.8" REQUIRED)
 endif ()
 
 if (NOT SKK_DEFAULT_PATH)
-    set(SKK_DEFAULT_PATH "/usr/share/skk/SKK-JISYO.L")
+    set(SKK_DEFAULT_PATH "/usr/share/skk/")
 endif ()
 
 # This will add a uninstall target. Comment out this line if other cmake packages
diff --git a/src/dictionary_list.in b/src/dictionary_list.in
index 76e47a5..bc2f60e 100644
--- a/src/dictionary_list.in
+++ b/src/dictionary_list.in
@@ -1,2 +1,6 @@
-type=file,file=@SKK_DEFAULT_PATH@,mode=readonly
+type=file,file=@SKK_DEFAULT_PATH@SKK-JISYO.L,mode=readonly
 type=file,file=$FCITX_CONFIG_DIR/kkc/dictionary,mode=readwrite
+type=file,file=@SKK_DEFAULT_PATH@SKK-JISYO.zipcode,mode=readonly
+type=file,file=@SKK_DEFAULT_PATH@SKK-JISYO.station,mode=readonly
+type=file,file=@SKK_DEFAULT_PATH@SKK-JISYO.jinmei,mode=readonly
+type=file,file=@SKK_DEFAULT_PATH@SKK-JISYO.geo,mode=readonly
